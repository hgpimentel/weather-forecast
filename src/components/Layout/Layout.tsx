import React from "react";
import { NavLink } from "react-router-dom";
import styled from "styled-components";
import ROUTE from "../../config/routes";

const Header = styled.header`
  width: 100%;
  height: 5rem;
  display: flex;
  justify-content: space-between;
  padding: 0 10%;
  align-items: center;
  background-color: #008080;
`;

const Nav = styled.nav`
  ul {
    list-style: none;
    display: flex;
    margin: 0;
    padding: 0;
  }

  li {
    margin-left: 1.5rem;
    font-size: 1.25rem;
  }

  a {
    text-decoration: none;
    color: #88dfdf;
  }

  a:hover,
  a:active,
  a.active {
    color: #e6fcfc;
  }
`;

const Logo = styled.div`
  font-size: 2rem;
  color: white;
`;

const Layout: React.FC = ({ children }) => {
  return (
    <>
      <Header>
        <Logo>Weather Forecast App</Logo>
        <Nav>
          <ul>
            <li>
              <NavLink to={ROUTE.TODAY}>Today</NavLink>
            </li>
            <li>
              <NavLink to={ROUTE.NEXT7DAYS}>Next 7 Days</NavLink>
            </li>
          </ul>
        </Nav>
      </Header>
      <main>{children}</main>
    </>
  );
};

export default Layout;
