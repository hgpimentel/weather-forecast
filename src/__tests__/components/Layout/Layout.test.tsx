import renderer from "react-test-renderer";
import Layout from "../../../components/Layout/Layout";
import { BrowserRouter } from "react-router-dom";

describe("Layout component", () => {
  it("renders correctly", () => {
    const component = renderer
      .create(
        <BrowserRouter>
          <Layout></Layout>
        </BrowserRouter>
      )
      .toJSON();

    expect(component).toMatchSnapshot();
  });
});
