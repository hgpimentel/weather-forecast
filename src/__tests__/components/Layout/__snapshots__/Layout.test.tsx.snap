// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component renders correctly 1`] = `
Array [
  <header
    className="sc-fujyAs fGuyzn"
  >
    <div
      className="sc-bdnxRM sc-jrsJWt eoiijl iedUoR"
    >
      <h1>
        Weather Forecast App
      </h1>
    </div>
    <nav
      className="sc-pNWdM ljLxKa"
    >
      <ul>
        <li>
          <a
            aria-current={null}
            href="/today"
            onClick={[Function]}
          >
            Today
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            href="/next7days"
            onClick={[Function]}
          >
            Next 7 Days
          </a>
        </li>
      </ul>
    </nav>
  </header>,
  <main />,
]
`;
